### ISSUE TITLE
camelCase JavaScript Asset Filenames

### ISSUE THREAD

**Issue by noi5e** [publiclab/plots2#9140]
See this [comment thread](https://github.com/publiclab/plots2/pull/9113#discussion_r568730680).

Some of the JS assets use `camelCase` naming convention, while others use `underscore_filename` convention:
<img width="294" alt="106628319-b9adcf00-658a-11eb-8af8-c780e5bcdc57" src="https://user-images.githubusercontent.com/4361605/107128576-a6914c80-6873-11eb-8ef0-908cf62e47fe.png">

Let's standardize them and use `camelCase` since that's the JS way! These make great first-timer-only issues. (I definitely welcome help in making these!)

Don't forget to do a search in your IDE for wherever the file is required. For example, some of these are loaded in `config/initializers/assets.rb`.
  
### Completed
- [x] `advanced_search.js` 
  - [x] 🔍 **[open issue #9141]**
  - [x] 🙋 **[claimed by @Hari-tn67]** 
  - [x] ❗ **[PR #9163]**
  - [x]  🎉  **[merged on 2/10!]**
- [x] `async_tag_subscriptions.js` 
  - [x] 🔍 **[open issue #9142]**
  - [x] 🙋 **[claimed by @katiebankhead on 2/13]** 
  - [x] ❗ **[PR #9194]**
  - [x] 🎉 **[merged on 2/14!]**
- [x] `atwho_autocomplete.js` 
  - [x] 🔍 **[open issue #9143]**
  - [x] 🙋 **[claimed by @shadowoflight on 2/13]**
  - [x] ❗ **[PR #9192]**
  - [x] 🎉 **[merged on 2/15!]**
- [x] `comment_expand.js` 
  - [x] 🔍 **[open issue #9144]**
  - [x] 🙋 **[claimed by @manchere on 2/16]** 
  - [x]  ❗ **[open PR #9206]**
  - [x]  🎉  **[merged on 2/16!]**
- [x] `header_footer.js`
  - [x] 🔍 **[open issue #9146]** 
  - [x] 🙋 **[claimed by @StellaWanja on 2/7]** 
  - [x] ❗ **[PR #9156]** 
  - [x]  🎉  **[merged on 2/9!]**
- [x] `leaflet_helper.js`
  - [x] 🔍 **[open issue #9147]** 
  - [x] 🙋 **[claimed by @redwan-mahmud on 2/11]** 
  - [x] ❗ **[PR #9147]**
  - [x] 🎉 **[merged on 2/15!]**
- [x] `main_image.js`
  - [x] 🔍 **[open issue #9148]** 
   - [x] 🙋 **[claimed by @larabee7777 on 2/11]** 
   - [x] ❗ **[open PR #9201]**
   - [x]  🎉  **[merged on 2/16!]**
- [x] `noty_notification.js`
  - [x] 🔍 **[open issue #9149]** 
  - [x] 🙋 **[claimed by @akshay4n on 2/21]**
  - [x] ❗ **[PR #9223]**
  - [x] 🎉 **[ready to merge!]** 
- [x] `restful_typeahead.js`
  - [x] 🔍 **[open issue #9150]**
  - [x] 🙋 **[claimed by @ghostoverflow on 2/21]**
  - [x] ❗ **[PR #9228]**
  - [x] 🎉 **[merged on 2/23!]** 
- [x] `submit_form_ajax.js`
  - [x] 🔍 **[open issue #9151]**
  - [x] 🙋 **[claimed by @StreakSharn on 2/7]** 
  - [x] ❗ **[PR #9157]**
  - [x]  🎉  **[merged on 2/9!]**
- [x] `textbox_expand.js`
  - [x] 🔍 **[open issue #9152]** 
  - [x] 🙋 **[claimed by @Nagsenwaghmare on 2/8]**  
  - [x] ❗ **[PR #9153]**
  - [x]  🎉  **[merged on 2/9!]**

### 🔬 Needs Further Research
- `simple-data-grapher.js`
  - This seems to be one of Public Lab's own external packages that we require in `package.json`?
  - May not be so easy to change this one, or out-of-scope for an FTO.

**Comment by larabee7777** [comment]
Hi do you need help here?

**Comment by noi5e** [comment]
@larabee7777 You've already completed one of these first-timer-only issues (#9148). I think generally each person gets only one FTO issue.

If you're looking for another way to contribute, you could browse through our [help wanted issues](https://github.com/publiclab/plots2/labels/help%20wanted). OR you could also help someone else get through their own FTO issue. maybe the two open issues above? You could leave a comment there seeing if they need any help.

You can also make your own FTO issue. We have a bunch of [first-timer-only candidate issues](https://github.com/publiclab/plots2/issues?q=is%3Aopen+is%3Aissue+label%3Afto-candidate) that need to be broken down and formatted into FTO issues.

**Comment by larabee7777** [comment]
> @larabee7777 You've already completed one of these first-timer-only issues (#9148). I think generally each person gets only one FTO issue.
> 
> If you're looking for another way to contribute, you could browse through our [help wanted issues](https://github.com/publiclab/plots2/labels/help%20wanted). OR you could also help someone else get through their own FTO issue. maybe the two open issues above? You could leave a comment there seeing if they need any help.
> 
> You can also make your own FTO issue. We have a bunch of [first-timer-only candidate issues](https://github.com/publiclab/plots2/issues?q=is%3Aopen+is%3Aissue+label%3Afto-candidate) that need to be broken down and formatted into FTO issues.

Ah ok I see thanks so much I will check out the other issues :)

**Comment by noi5e** [comment]
Okay, I think with the two PRs that got merged today, we can close this issue!!! Thanks to everyone who made their first PRs and helped us with this! @Hari-tn67 @katiebankhead @shadowoflight @manchere @StellaWanja @redwan-mahmud @larabee7777 @akshay4n @ghostoverflow @StreakSharn @Nagsenwaghmare 🏆 🎉 🍾 

If any of you would like to help out some more, or get more experience in open source, check out [https://code.publiclab.org/](https://code.publiclab.org/). Feel free to check out our [help wanted issues](https://github.com/publiclab/plots2/issues?q=is%3Aopen+is%3Aissue+label%3A%22help+wanted%22), create your own first-timer-only issue, do code review... You can also ask for help setting up `plots2` locally, try to find bugs, or write issues. And if you haven't checked us out on Gitter, please do!
